src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -395,6 +395,7 @@ private void AddNameAndTypeArgumentsOrParameters(INamedTypeSymbol symbol)
                 AddDelegateParameters(symbol);
             }
 
+            AddUnderlyingEnumType(symbol);
```

Comment:
Adding this here assumes, that the underlying enum type is treated more or less like type parameters.